### DISCUSSION TITLE
Button Image not resizing with bounds

### DISCUSSION THREAD

**Discussion by thamEnJie** [sst-inc/help#43] [Q&A - iOS] [answered]
Hi, 

this may be really stupid and simple, but I search the web and did what they said to do (assuming I followed them correctly). I thought it was normal to not have it resize ith bounds until I watched some video and I legit have no idea if this is normal or not.

Thanks :)

**Comment by Joeispro99** [comment, marked as answer]
Can you send the project or at least some screenshots of expectations and reality.

**Reply by thamEnJie** [reply]
Thanks for the prompt reply! :D

For me, it is [like this](https://user-images.githubusercontent.com/89586587/137591988-8063f510-c69f-4c50-bdbe-df21440c5b8f.mov)

But in the video I watched, I was [like this](https://user-images.githubusercontent.com/89586587/137591920-186c5e07-862d-4a82-ad3f-56597e44bf3f.mov)

(sorry I thought I replied I sent wrongly)

**Reply by granwyntan** [reply]
Hi @thamEnJie, maybe u could try to set **Clip To Bounds** in the identity inspector or set `clipsToBounds = true` in your code.
As for resizing, there are multiple options, which u can select the **Content Mode** in the identity inspector or set it to `scaleToFill`, `scaleAspectFit` or `scaleAspectFill` in your code. Personally, I recommend `scaleAspectFit` as the aspect ratio will be kept no matter how much the image is scaled. Hope it helps!

**Reply by thamEnJie** [reply]
Hmm, now I am even more confused. I opened my old project (cos I am redoing it) that is in the bin, and now I can resize the cookie?? I thought I could not do it before...

**Reply by thamEnJie** [reply]
> Hi @thamEnJie, maybe u could try to set **Clip To Bounds** in the identity inspector or set `clipsToBounds = true` in your code. As for resizing, there are multiple options, which u can select the **Content Mode** in the identity inspector or set it to `scaleToFill`, `scaleAspectFit` or `scaleAspectFill` in your code. Personally, I recommend `scaleAspectFit` as the aspect ratio will be kept no matter how much the image is scaled. Hope it helps!

Hello @granwyntan :D 

Hmm, I tried this for a second time and it still isn't working... It only clips it..

much appreciated

**Reply by thamEnJie** [reply]
It may be because of a new update to xcode 13... I might be wrong but there are some discrepancies in the UI... (which may not make sense)

And I realised I used the background in the old one, which from some stack overflow post said that it resizes with the button. I can't find the background option in the new one...

Old:
<img width="241" alt="Screenshot" src="https://user-images.githubusercontent.com/89586587/137592960-496b6aba-ea21-4632-8107-e18efc0c4902.png">
<img width="257" alt="Screenshot" src="https://user-images.githubusercontent.com/89586587/137592999-c96e888c-9d6c-4d68-a886-d26941d52e43.png">


New:
<img width="236" alt="Screenshot" src="https://user-images.githubusercontent.com/89586587/137592930-61aa863a-92a7-45c1-ac30-c10a0f1202df.png">
<img width="274" alt="Screenshot" src="https://user-images.githubusercontent.com/89586587/137592983-afa43944-f010-46e7-b0f0-de56ad634e5a.png">


going to check if I messed up the first few settings :D

**Reply by granwyntan** [reply]
Hi @thamEnJie for the View > Content Mode u set it to Scale To Fill, I think setting it to Aspect Fit would resolve your issue
<img width="222" alt="Screenshot 2021-10-16 at 23 35 00" src="https://user-images.githubusercontent.com/61010627/137593469-ca6d8171-ef35-4f08-b3c1-3cf172abd18c.png">

**Reply by granwyntan** [reply]
Also check Clip To Bounds under drawing
<img width="265" alt="Screenshot 2021-10-16 at 23 37 22" src="https://user-images.githubusercontent.com/61010627/137593523-713e81d9-bebd-48d1-8304-6a21af97dd7a.png">

**Reply by thamEnJie** [reply]
Hi @granwyntan, have tried it before.. and it still has not worked... ;-;

<img width="541" alt="Screenshot" src="https://user-images.githubusercontent.com/89586587/137593615-55c0bd7c-f825-45af-b0d3-ef830feccd7b.png">

**Reply by thamEnJie** [reply]
Ah! I found the answer from here: [https://stackoverflow.com/questions/69439301/uibutton-image-behavior-changed-in-ios-15](https://stackoverflow.com/questions/69439301/uibutton-image-behavior-changed-in-ios-15)

So in accordance to what was said about the new look, it was because of the new system and stuff. Now there is plain and more.

So I just needed to change the style back to default to get the old settings 🙃 

Sorry for wasting both of your time.. And much thanks for your help too! @granwyntan @Joeispro99

**Reply by granwyntan** [reply]
Great, glad that your issue has been resolved :)

**Reply by thamEnJie** [reply]
Thanks!

okay cool now my cookie became blue. lol

**Comment by granwyntan** [comment]
Lol

**Reply by thamEnJie** [reply]
fixed woohoo time to sleep. please sleep too :D
